src/EnergyPlus/HeatingCoils.cc

```diff
@@ -2132,8 +2130,6 @@ namespace HeatingCoils {
                 EffHS = HeatingCoil(CoilNum).MSEfficiency(StageNumHS);
```

Comment:
`auto &heatingCoil = state.dataHeatingCoils->HeatingCoil(CoilNum);` is a much more useful shortcut than `auto &HeatingCoil = state.dataHeatingCoils->HeatingCoil;` in this case. In general, try to have references to objects rather than to arrays of objects.

Reply:
Also, it looks like `CoilNum` is only read in this function and not written, so it should be passed by value rather than reference.

Reply:
Thanks, I will improve the codes with your comments.